pkg/controller/connectivitycheck/connectivity_check_controller.go

```diff
@@ -0,0 +1,404 @@
+	if err != nil {
+		recorder.Warningf("EndpointDetectionFailure", "unable to determine openshift-network-diagnostics network-check-target endpoints: %v", err)
+		return nil
+	}
+
+	for _, address := range addresses {
+		templates = append(templates, NewPodNetworkConnectivityCheckTemplate(net.JoinHostPort(address.hostName, address.port), "openshift-network-diagnostics", withTarget("network-check-target", strings.Split(address.nodeName, ".")[0])))
+	}
+	return templates
+}
+
+// listAddressesForGenericPodServiceEndpoints returns network-check-target service endpoints ip
+func (c *connectivityCheckTemplateProvider) listAddressesForGenericPodServiceEndpoints(recorder events.Recorder) ([]endpointInfo, error) {
+	var results []endpointInfo
+	endpoints, err := c.diagnosticsEndpointsLister.Endpoints("openshift-network-diagnostics").Get("network-check-target")
+	if err != nil {
+		return nil, err
+	}
+	for _, subset := range endpoints.Subsets {
+		for _, address := range subset.Addresses {
+			for _, port := range subset.Ports {
+				results = append(results, endpointInfo{
+					hostName: address.IP,
+					port:     strconv.Itoa(int(port.Port)),
+					nodeName: *address.NodeName,
+				})
+			}
+		}
+	}
+	return results, nil
+}
+
+func (c *connectivityCheckTemplateProvider) getTemplatesForAPILoadBalancerChecks(recorder events.Recorder) []*v1alpha1.PodNetworkConnectivityCheck {
+	var templates []*v1alpha1.PodNetworkConnectivityCheck
+	infrastructure, err := c.infrastructureLister.Get("cluster")
+	if err != nil {
+		recorder.Warningf("EndpointDetectionFailure", "error detecting api load balancer endpoints: %v", err)
+		return nil
+	}
+
+	apiUrl, err := url.Parse(infrastructure.Status.APIServerURL)
+	if err != nil {
+		recorder.Warningf("EndpointDetectionFailure", "error detecting external api load balancer endpoint: %v", err)
+
+	} else {
+		templates = append(templates, NewPodNetworkConnectivityCheckTemplate(apiUrl.Host, "openshift-network-diagnostics", withTarget("load-balancer", "api-external")))
+	}
+
+	apiInternalUrl, err := url.Parse(infrastructure.Status.APIServerInternalURL)
+	if err != nil {
+		recorder.Warningf("EndpointDetectionFailure", "error detecting internal api load balancer endpoint: %v", err)
+	} else {
+		templates = append(templates, NewPodNetworkConnectivityCheckTemplate(apiInternalUrl.Host, "openshift-network-diagnostics", withTarget("load-balancer", "api-internal")))
+	}
+	return templates
+}
+
+type endpointInfo struct {
+	hostName string
+	port     string
+	nodeName string
+}
+
+func withTarget(label, target string) func(check *v1alpha1.PodNetworkConnectivityCheck) {
+	return WithTarget(label + "-" + target)
+}
+
+func Start(ctx context.Context, kubeConfig *rest.Config) error {
+	protoKubeConfig := rest.CopyConfig(kubeConfig)
+	protoKubeConfig.ContentType = "application/vnd.kubernetes.protobuf,application/json"
+	eventRecorder := &eventrecorder.LoggingRecorder{}
+	kubeClient, err := kubernetes.NewForConfig(protoKubeConfig)
```

Comment:
you can get this from `pkg/client`

Reply:
it seems there are methods left to implement:

pkg/controller/connectivitycheck/connectivity_check_controller.go:387:70: cannot use kubeClient (type *"github.com/openshift/cluster-network-operator/pkg/client".Client) as type kubernetes.Interface in argument to NewNetworkConnectivityCheckController:
        *"github.com/openshift/cluster-network-operator/pkg/client".Client does not implement kubernetes.Interface (missing AdmissionregistrationV1 method)

Given the short time to deliver, let's refactor later.